devs/docs/architecture.rst

```diff
@@ -0,0 +1,100 @@
+=====================
+CrateDB Architecture
+=====================
+
+CrateDB is a distributed SQL database. From a high-level perspective, the
+following components can be found in each node of CrateDB:
+
+.. image:: ../resources/architecture.svg
+
+Components
+-------------
+
+SQL Engine
+............
+
+CrateDB's heart consists of a SQL Engine which takes care of parsing SQL
+statements and executing them in the cluster.
+
+The engine is comprised of the following components:
+
+1. Parser: Breaks down the SQL statement into its components.
+2. Analyzer: Performs semantic processing of the statement including
+   verification, type annotation, and optimization.
+3. Planner: Builds an execution plan from the analyzed statement.
+4. Executor: Executes the plan in the cluster and collects results.
+
+Class entry point:
+
+- `SqlParser`
+- `RelationAnalyzer`
+- `Planner`
+- `LogicalPlan` / `Plan`
+- `NodeOperationTree`
+- `ExecutionPhasesTask`
+- `BatchIterator`
+
+Input
+.....
+
+When you first use CrateDB, you probably want to check out its Web
```

Comment:
I'd assume that the reader has some familiarity with CrateDB from a user perspective. Otherwise the gap here from basic introduction to code seem quite large

Reply:
Yes, it is very hard to imagine someone contributing without having used CrateDB at all.

Reply:
That's why I'd even shorten the web-ui introduction here. Instead I'd state in the introduction that it's expected that the reader is familiar with the functionality of CrateDB and link to the user documentation.

Anyway - these are just some thoughts / inputs, because I felt like being picked up very gentle just to be dropped into the deep water afterwards. The mental gap being "There is an admin ui" and "here are two components which are part of the protocol implementation" is rather large. But to be honest I've also no good idea how to soften this gap without getting lost in all the details.

This applies generally to the whole document as it is. Personally I feel that the narrative contains too many big leaps. But I'm probably also not the best person to make this judgement.

I think it's fine to get this in if we see this is a starting point. But if you want to iterate on it I'd recommend to have some more people read it and talk to them to see what kind of questions pop up.

Reply:
I think it is fine to have leaps in this document. It is just to have an overview of what components exists, alongside with some pointers to the implementation.

I've tried to get feedback from more people. Will try harder ;)

Reply:
I've removed the admin ui information and added a pointer to the getting started section in the beginning.